Redevelop the Inspector widget

bennothommo opened the pull request:
Please see #498 for more information on this PR, and keep this thread only for discussing the technical implementation.

<a href="https://gitpod.io/#https://github.com/wintercms/winter/pull/499"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

bennothommo committed:
Initial work on Inspector rewrite

bennothommo committed:
Use correct API variable in Mix webpack fixture

bennothommo committed:
Tweak overlay to always fade, rename plugin

bennothommo committed:
Establish code style for Vue widgets

bennothommo committed:
Add initial styling and base functionality of Inspector

LukeTowers commented on modules/system/console/fixtures/mix.webpack.js.fixture:
```diff
@@ -11,26 +11,26 @@ module.exports = async () => {
     mix.listen('init', function (mix) {
         // Setup Winter path aliases
-        mix._api.alias({
+        mix.api.alias({
```
What's the reasoning for this change?

bennothommo replied:
I was getting errors using `mix:watch` as soon as I introduced Vue compiling. Turns out that `api` is the correct variable to use for this, and `_api` is the internal reference. The `api` variable actually creates the `_api` variable if it doesn't exist, so I believe that perhaps when Vue compiling is used, the Mix initialisation happens earlier.

LukeTowers replied:
Huh. Is this something that we should fix in a main branch right now?

bennothommo replied:
I'd say so, but I've just done it here for now so I can be sure it's working.

bennothommo committed:
Initial work on config processing, further styling, hide function

bennothommo committed:
Tweak styling of overlay

bennothommo committed:
Styling inspector, adding initial groundwork for fields

bennothommo committed:
Convert Mix config into a global "System" config

Since we'll likely be using the same libraries across widgets (ie. Vue), we'll instead apply everything to the System module and extract it out on build, so the individual modules should be very smal.

bennothommo committed:
Use a (system) global ESLint config and Gitignore

bennothommo committed:
Fix Code Quality checks after refactor

bennothommo committed:
Recompile assets with Snowboard no longer extracted

bennothommo committed:
Include vendor files in JS tests

bennothommo committed:
Compile system assets together

Similar to the original framework, the UI and inspector widgets were combined into one framework file.

bennothommo committed:
Remove old vendor files

bennothommo committed:
Fix tests

bennothommo committed:
Merge branch 'develop' into wip/inspector-rewrite

bennothommo committed:
Re-organise Inspector styles, add comment icon

bennothommo committed:
Initial tooltip plugin WIP

bennothommo committed:
Recompile assets

bennothommo committed:
Merge remote-tracking branch 'origin/develop' into wip/inspector-rewrite

bennothommo committed:
Fix build

bennothommo committed:
Add linting command

bennothommo committed:
Linting, renaming base UI elements

bennothommo committed:
WIP initial field functionality and management

bennothommo committed:
Add tooltip support, add hover effect

bennothommo committed:
Adjust way configuration is read for an Inspector widget

This should be backwards compatible in that any classes who have `onGetInspectorConfiguration` methods will still use their overridden method, but for new applications, they can now use an `inspector.yaml` file within the class name for the inspector, or the controller itself.

It should also support the old `configuration` based array structure, as well as the new one without this key.

bennothommo committed:
Force special text fields to always appear as text fields

jaxwilko commented:
@what-the-diff 🤔

bennothommo committed:
WIP secondary form, field inputs

bennothommo committed:
Recompile

bennothommo committed:
Merge branch 'develop' into wip/inspector-rewrite

bennothommo committed:
Allow cms.restrictBaseDir to be controlled via env variable

bennothommo committed:
Fix focus state, add field types, tweaks

bennothommo committed:
Add checkbox field

bennothommo committed:
Recompile
